sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java

```diff
@@ -719,7 +906,9 @@ private void checkArguments() {
                           return getUserDataMapper().mapRow(element);
                         }
                       }))
-              .apply("Map Objects array to CSV lines", ParDo.of(new MapObjectsArrayToCsvFn()))
+              .apply(
+                  "Map Objects array to CSV lines",
+                  ParDo.of(new MapObjectsArrayToCsvFn(getQuotationMark())))
```

Comment:
I know this was discussed earlier, but I am somewhat concerned about using CSV as the method to write. Is this not risky? Have you verified that it works well for all types (Numeric, Date/Time, Decimal, Strings with newlines, etc.)?

I recommend you add an integration test that tests all - or most - data types once we have the Snowflake instance to test.

Reply:
Yes, I agree that CSV can be a little bit problematic here. On the other hand apparently it is the most efficient way of loading data to Snowflake.
We have some tests that were checking different ways of data quotation and problematic characters. But I agree it should be included in [integration tests](https://github.com/apache/beam/blob/master/sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/BatchSnowflakeIOIT.java) as well. Would it be ok with you if I create Jira ticket for extending integration tests?